### DISCUSSION TITLE
How do I get to intellisense work?

### DISCUSSION THREAD

**Discussion by ghost** [pixijs/pixijs#7821] [Q&A] [answered]
I'm trying to get intellisense to work with PixiJS but I'm having some trouble, so far I have only tried this which looks like it should work but it isn't for me.

1.  Install PixiJS through NPM: `npm install pixi.js`
2. import pixi into my JS file: `import * as PIXI from 'pixi.js';`

[Where I'm getting these steps from.](https://www.html5gamedevs.com/topic/30187-visual-studio-code-intellisense-code-completion/?tab=comments#comment-173408)
[Here is a github repo of everything so its easier to see where I'm messing up.](https://github.com/ReallPanda/pixiJS)
Also please bear with me on this, I'm completely new to importing libraries so I could be doing something completely wrong without knowing.

**Comment by miltoncandelero** [comment, marked as answer]
I am not sure how javascript intellisense works but if you are willing to try typescript I can offer you https://www.pixijselementals.com/

It will help you setup a coding environment with npm, pixijs, and typescript. 🤔

**Reply by ghost** [reply]
awesome thank you so much!

**Reply by domis86** [reply]
those docs/examples seems useful :+1:  thanks

**Reply by mmdevcodes** [reply]
@miltoncandelero thank you for making this! Unrelated, but do you have an opinion on creating user interfaces? For example menus, hud, etc? Should it be done manually or do you think there's a good library for doing this in pixi?

**Reply by miltoncandelero** [reply]
I usually create my own but I also know that https://github.com/pixijs/pixi-ui exists and some people do stuff with react on the dom or even with react-like syntax inside pixi.

Pick your poison 😄
